src/advection_stencils.jl

```diff
@@ -203,3 +203,118 @@ end
 
 " Return the left and right stencil size of the second-order upwind stencil. "
 stencil_size(s::typeof(upwind2_stencil)) = (2, 2)
+
+
+
+"""
+$(SIGNATURES)
+
+PPM advection in 1-D with changing grid sizes (Collela and Woodward, 1984)
+
+* ϕ is the scalar field at the current time step, it should be a vector of length 8 (3 cells on the left, the central cell, and 4 cells on the right).
+* U is the velocity at both edges of the central grid cell, it should be a vector of length 2.
+* Δt is the length of the time step.
+* Δz is the grid spacing. Here Δz is a vector rather than a single float number because Δz varies over grid cells.
+
+The output will be time derivative of the central index (i.e. index 4)
+of the ϕ vector (i.e. dϕ/dt).
+
+(The output is dependent on the Courant number, which depends on Δt, so Δt needs to be
+an input to the function.)
+"""
+function ppm_stencil_grid(ϕ, U, Δt, Δz; kwargs...)
```

Comment:
I think we want to avoid having to different stencils, one where the grid is variable and one where it isn't, because that may be confusing for the user and also results in more code to maintain.

I think in this case it will be better to just change the code for `ppm_stencil` so that it takes a vector of delta z values and works for the variable grid case, rather than having 2 seperate functions